src/Model/AuditReaderInterface.php

```diff
@@ -19,46 +19,28 @@
 interface AuditReaderInterface
 {
     /**
-     * @param string $className
-     * @param string $id
-     * @param string $revision
-     *
-     * @return object
+     * @param mixed $id
      */
-    public function find($className, $id, $revision);
+    public function find(string $className, $id, int $revision): ?object;
```

Comment:
not sure if we should consider $revision as mixed too, to avoid problems, like with $id. The EntityAuditBundle uses ints but that's an implementation detail. Thoughts?

Reply:
I'm not sure if we have another implementation of this interface. 
This seems to be made especially for the audit bundle...

It's not clear to me the difference between
```
find(string $className, $id, int $revision)
```
and
```
findRevision(string $className, $id, int $revision)
```

For the $revision argument, int seems logic at first sight: we want the first one, the second one, ...
If I understand correctly, there is always an order for the revisions.

Reply:
(Disclaimer: I am not a user of this AuditReader or EntityAuditBundle)

It seems like the find method, returns the object for that concrete revision ID. And the findRevision returns the Revision object https://github.com/simplethings/EntityAuditBundle/blob/master/src/SimpleThings/EntityAudit/Revision.php

Reply:
I think @phansys use this feature.

Maybe he'll have a point of view about the `int` typehint of the revision.

@jordisala1991 If it's a `revision id` and we change it to mixed, it could be a nice idea to rename `$revision` to `$revisionId`.

Reply:
Just found something.

This feature is 100% related to the EntityAuditBundle, since the revision is used this way in the template
https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Resources/views/CRUD/base_history.html.twig

I see three solution
- Deprecate/remove this feature, since the bundle is not maintained. But it would be sad for the current users.
- Move this feature/route/template/interface/... to storage bundle. But I don't know how hard it is.
- Create our own Revision interface. DoctrineORM could create a SonataRevision from the current Revision.

WDYT @jordisala1991 @phansys ?

If you agree we can create an issue.

Reply:
I'm heavily using the audit feature in my projects, so I'd like to keep its support regardless the implementation.
I think an issue tracking this will be fine.

Reply:
> I'm heavily using the audit feature in my projects, so I'd like to keep its support regardless the implementation.

Sure, that's why I was thinking the third solution was the good one.

Reply:
Made https://github.com/sonata-project/SonataAdminBundle/issues/6279

Reply:
@jordisala1991 I think, the only change to make here would be to replace 
```
int $revision
```
by 
```
mixed $revisionId
```

Reply:
Done

Reply:
I proposed to also rename the params.
```
find(string $className, $id, $revisionId): ?object;
findRevision(string $className, $revisionId): ?object;
diff(string $className, $id, $oldRevisionId, $newRevisionId): array;
``

`$revisionId` is clearer than `$revision` which let think it can be a `Revision` entity.

Reply:
Done!